### PR TITLE
fix todo in disruption.go

### PR DESCRIPTION
**Pull request by islinwb**
**What this PR does / why we need it**:
fix todo

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by deads2k** [comment]
> /assign @deads2k @sttts

The individual controller logic is generally owned by apps, not apimachinery.  The repercussions of making changes like this are not always obvious.

/unassign
/assign @davidopp 

@davidopp I see disruption and think of you, but I'm not sure which sig has it.  @kubernetes/sig-scheduling-pr-reviews  maybe?

File: pkg/controller/disruption/disruption.go [hunk 1]
```diff
@@ -311,10 +311,13 @@ func (dc *DisruptionController) addDb(obj interface{}) {
 }
 
 func (dc *DisruptionController) updateDb(old, cur interface{}) {
-	// TODO(mml) ignore updates where 'old' is equivalent to 'cur'.
-	pdb := cur.(*policy.PodDisruptionBudget)
-	glog.V(4).Infof("update DB %q", pdb.Name)
-	dc.enqueuePdb(pdb)
+	oldPdb := old.(*policy.PodDisruptionBudget)
+	curPdb := cur.(*policy.PodDisruptionBudget)
+	if curPdb.ResourceVersion == oldPdb.ResourceVersion {
```
Comment (k82cn) [inline pkg/controller/disruption/disruption.go]: IMO, the `TODO` means pdb's spec instead of version.

Comment (kevin-wangzefeng) [reply pkg/controller/disruption/disruption.go]: add an import `apiequality "k8s.io/apimachinery/pkg/api/equality"`
and you can use `apiequality.Semantic.DeepEqual` to compare the two specs here
```go
if !apiequality.Semantic.DeepEqual(curPdb.Spec, oldPdb.Spec) {
```

Comment (islinwb) [reply pkg/controller/disruption/disruption.go]: @k82cn @kevin-wangzefeng Thanks for reviewing!

**Pull request closed without merging**